src/Microsoft.AspNetCore.OData/ODataController.cs

```diff
@@ -1,36 +1,55 @@
 ﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
 // Licensed under the MIT License.  See License.txt in the project root for license information.
 
-using System;
 using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Results;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.OData.Edm;
-using Microsoft.OData.Edm.Csdl;
 
 namespace Microsoft.AspNet.OData
 {
     /// <summary>
     /// Defines a base class for OData controllers that support writing and reading data using the OData formats.
     /// </summary>
+    /// <remarks>These attributes and signature uses types that are AspNetCore-specific.</remarks>
+    [ODataFormatting]
+    [ODataRouting]
     [ApiExplorerSettings(IgnoreApi = true)]
     public abstract partial class ODataController : ControllerBase
```

Comment:
do we still need "ODataController"?

in Asp.net core, base controller class is not necessary.

Reply:
It's used in AspNet to inject formatting and routing, which is no longer needed in AspNetCore. However, ODataController does provide some result classes and it makes migrating/documentation easier in that creating a controller is the same in both platforms.